File: 0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char,int>hm;
        for(char i:magazine){
            hm[i]++;
        }
        for (char i:ransomNote){
            if (hm[i]==0){
                return false;
            }
            hm[i]--;

        }return true;
    }
};

File: 0383-ransom-note/0383-ransom-note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        HashMap<Character,Integer>hm=new HashMap<>();
        for (char i:magazine.toCharArray()) {
            hm.put(i,hm.getOrDefault(i,0)+1);
        }
        for (char i:ransomNote.toCharArray()) {
            if (!hm.containsKey(i)||hm.get(i)==0) {
                return false;
            }
            hm.put(i, hm.get(i)-1);
        }
        return true;
    }
}
